packages/components/pie-cookie-banner/test/component/pie-cookie-banner.spec.ts

```diff
@@ -148,5 +160,70 @@ test.describe('PieCookieBanner - Component tests', () => {
 
         expect(modal).not.toBeVisible();
         expect(cookieBanner).not.toBeVisible();
+
+        const [expectedCookieBannerPrefsSavedEvent] = preferences.filter(({ id }) => id !== 'all')
+        .map(({ id, isChecked }) => ({ [id]: !!isChecked }));
+        expect(cookieBannerPrefsSavedEvent).toMatchObject(expectedCookieBannerPrefsSavedEvent);
+    });
+
+    test('should always set the `necessary` preference to on and disabled states', async ({ mount, page }) => {
+        // Arrange
+        await mount(PieCookieBanner, {
+            props: {} as CookieBannerProps,
+        });
+
+        // Act
+        await page.click(managePreferencesSelector);
+
+        // Assert
+        const preference = await page.locator(getPreferenceItemSelector('necessary')).isChecked();
+        expect(preference).toBe(true);
+        expect(preference).toBe(true);
```

Comment:
is this double assertion intentional? I think we need to assert the disabled state once and the checked state once :)